Khan/live-editor
# Remove visibly, replace with Page Visibility API

### High-level description of change

This replaces the visibly submodule with the Page Visibility API, now supported in all relevant browsers. See: https://caniuse.com/#feat=pagevisibility

### Are there performance implications for this change?

One less module, so smaller size! (Also, visibly was likely polling. Native is better).

### Have you added tests to cover this new/updated code?

No, I ran previous tests and ran the manual test.

### Risks involved

There could be a subtle difference between visibly and the visibility API? Hopefully not, since visibly was a polyfill.

### Are there any dependencies or blockers for merging this?

No.

### How can we verify that this change works?
- copy/paste the example from #260 into the demo app
- hide the browser or switch to a different tab and wait for ~2 minutes
- return to the demo page and verify that the script has continued to run uninterrupted

Taken from the original visibly PR: https://github.com/Khan/live-editor/commit/6b3b5e505179649c1a291d1dcfe76872b463ee0a
